core/src/main/java/com/sequenceiq/cloudbreak/service/identitymapping/ObjectStorageValidateRequestDecorator.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.service.identitymapping;
+
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.AWS;
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.AZURE;
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.GCP;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageValidateRequest;
+import com.sequenceiq.common.api.cloudstorage.AccountMappingBase;
+
+@Component
+public class ObjectStorageValidateRequestDecorator {
+
+    @Inject
+    private AwsMockAccountMappingService awsMockAccountMappingService;
+
+    @Inject
+    private AzureMockAccountMappingService azureMockAccountMappingService;
+
+    @Inject
+    private GcpMockAccountMappingService gcpMockAccountMappingService;
+
+    public void decorateWithMockAccountMapping(ObjectStorageValidateRequest request) {
+        if (request.getCloudStorageRequest().getAccountMapping() == null) {
+            Map<String, String> groupMappings = null;
+            Map<String, String> userMappings = null;
+            String adminGroupName = request.getMockAccountMappingSettings().getAdminGroupName();
+            switch (request.getCloudPlatform()) {
+                case AWS:
+                    if (adminGroupName != null) {
```

Comment:
The code on lines 35-42 are almost identical in the three cases. It could make sense to extract them to a method and send as parameter the 2 varying parameters